File: tools/http/test-client/java/src/main/java/io/opentelemetry/conformance/http/ContractError.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */
package io.opentelemetry.conformance.http;

/** A server answered something the contract does not describe. */
public final class ContractError extends RuntimeException {

  private static final long serialVersionUID = 1L;

  ContractError(String message) {
    super(message);
  }

  ContractError(String message, Throwable cause) {
    super(message, cause);
  }
}

File: tools/http/test-client/js/src/contract-error.js
// Copyright The OpenTelemetry Authors
// SPDX-License-Identifier: Apache-2.0

"use strict";

/** A server answered something the contract does not describe. */
class ContractError extends Error {
  constructor(message, options) {
    super(message, options);
    this.name = "ContractError";
  }
}

module.exports = { ContractError };
